scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+        raise RuntimeError(
+            f'Error in get columns from {tablename}. Restart sparkSession and try again'
+        ) from e
+
+    if method == 'dbconnect' and sparkSession:
+        print('processes = ', processes)
+        df = sparkSession.table(
+            tablename)  # "main.tpcds_sf100_delta.store_sales")
+
+        # Running the query and collecting the data as arrow or json.
+        signed, _, _ = df.collect_cf('arrow')  # pyright: ignore
+        print(f'len(signed) = {len(signed)}')
+
+        args = get_args(signed, json_output_path, columns, cf_collect_type)
+
+        # Stopping the SparkSession to avoid spilling connection state into the subprocesses.
+        sparkSession.stop()
+
+        with ProcessPoolExecutor(max_workers=processes) as executor:
+            list(executor.map(download_starargs, args))
+
+    elif method == 'dbsql' and cursor:
+        for start in range(0, nrows, batch_size):
+            end = min(start + batch_size, nrows)
+            fetch_data(method, cursor, sparkSession, start, end, order_by,
+                       tablename, columns_str, json_output_path)
+
+    if cursor is not None:
+        cursor.close()
+
+
+def fetch_DT(args: Namespace) -> None:
+    """Fetch UC Delta Table to local as jsonl."""
+    log.info(f'Start .... Convert delta to json')
+
+    obj = urllib.parse.urlparse(args.json_output_path)
+    if obj.scheme != '':
+        raise ValueError(
+            f'Check the json_output_path and verify it is a local path!')
+
+    if os.path.exists(args.json_output_path):
+        if not os.path.isdir(args.json_output_path) or os.listdir(
+                args.json_output_path):
+            raise RuntimeError(
+                f'A file or a folder {args.json_output_path} already exists and is not empty. Remove it and retry!'
+            )
+
+    os.makedirs(args.json_output_path, exist_ok=True)
+
+    log.info(f'Directory {args.json_output_path} created.')
+
+    method = ''
+    dbsql = None
+    sparkSession = None
+
+    if hasattr(args, 'http_path') and args.http_path:
+        method = 'dbsql'
+        try:
+            dbsql = sql.connect(
+                server_hostname=args.DATABRICKS_HOST,
+                http_path=args.http_path,
+                access_token=args.DATABRICKS_TOKEN,
+            )
+        except Exception as e:
+            raise RuntimeError(
+                'Failed to create sql connection to db workspace. Check {server_hostname} and {http_path} and access token!'
+            ) from e
+    else:
+        method = 'dbconnect'
+        if not args.cluster_id:
+            session_id = str(uuid4())
+            sparkSession = DatabricksSession.builder.host(
+                args.DATABRICKS_HOST).token(args.DATABRICKS_TOKEN).header(
+                    'x-databricks-session-id', session_id).getOrCreate()
+        else:
+            # IMPORTANT: make sure cluster has runtime newer than 14.1.0, the databricks-connect client version.
```

Comment:
the comment disagrees with the min DBR version constant at the top of the file, which is correct?

Reply:
Good catch! I changed the one on the top to 14.0.0 by mistake. 14.1.0 is the minimum.